web/packages/design/src/Popover/Transition.tsx

```diff
@@ -16,22 +16,19 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React, { useLayoutEffect } from 'react';
 
-class Transition extends React.Component {
-  componentDidMount() {
-    const node = ReactDOM.findDOMNode(this);
-    this.props.onEntering(node);
-  }
-
-  render() {
-    const { children, ...childProps } = this.props;
-    delete childProps.onEntering;
-
-    const child = React.Children.only(children);
-    return React.cloneElement(child, childProps);
-  }
+export function Transition({
+  onEntering,
+  children,
+}: {
+  children: React.ReactNode;
+  onEntering: () => void;
+}) {
```

Comment:
```suggestion
export function Transition({
  onEntering,
  children,
}: PropsWithChildren<{
  onEntering: () => void;
}>) {
```

Lets use the React types, in case it ever changes for whatever reason